### PR TITLE
feat!: change default plugin dir

### LINKED ISSUE #997: :four_leaf_clover: `Proposal`: Specify the Plugin Directory Through the Configuration file
## What Would You Like to Add? Why Is This Needed?

We have a `-d` flag to specify the plugin directory, and the default value is `.devstream`. I believe this flag doesn't play a value and almost everyone is using its default behavior.

I believe that using `dtm` in different directories is a very generic requirement, for example, if today I finish `init` in `/tmp` and tomorrow I run `dtm` again under `/root`, I want `dtm` to find the plugins I downloaded yesterday, not every time I tell `dtm` where the plugins are, or I can only run `dtm` in a specific directory.

## Design

1. delete `-d` flag;
2. add `pluginDir` in core config. e.g.

```yaml
# core config
varFile: "" # If not empty, use the specified external variables config file
toolFile: "" # If not empty, use the specified external tools config file
pluginDir: "" If empty, use the "/usr/local/devstream/plugin" directory as default
state: # state config, backend can be local or s3
  backend: local
  options:
    stateFile: devstream.state
```

## Anything else

Done-Done Check

- [ ] All related documents and yaml files have been updated
- [ ] Feature has been implemented
- [ ] Feature has been tested and works properly

### PR DESCRIPTION
**Pull request by iyear**
Signed-off-by: iyear <ljyngup@gmail.com>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [ ] I have added relevant tests

## Description
A simpler solution to the plugin absolute path problem that does not affect documentation.

The only impact is that the developer needs to specify `-d .devstream` when debugging the plugin.

## Related Issues
close #997 

## New Behavior (screenshots if needed)
The plugin version shown is for demonstration purposes only, not the actual situation.

linux
windows
dev-plugin

### CODE REVIEW

**Comment by iyear** [comment]
@devstream-io/reviewer 
## New Behavior
only set flag:

set-flag

only set config:

set-config

default value:

set-default

set both flag and config(flag has higher priority):

set-both-flag-config

File: cmd/devstream/init.go [hunk 1]
```diff
@@ -3,11 +3,12 @@ package main
 import (
 	"fmt"
 
+	"github.com/devstream-io/devstream/pkg/util/file"
```
Comment (daniel-hutao) [inline cmd/devstream/init.go:6]: goimports

File: docs/core-concepts/config.md [hunk 2]
```diff
@@ -21,6 +22,8 @@ varFile: variables.yaml
 
 toolFile: tools.yaml
 
+pluginDir: /usr/local/.devstream/plugins # optional
```
Comment (daniel-hutao) [inline docs/core-concepts/config.md:25]: I think `/usr/local/devstream/plugins` is better here.

File: internal/pkg/pluginengine/cmd_apply.go [hunk 3]
```diff
@@ -4,6 +4,8 @@ import (
 	"errors"
 	"os"
 
+	"github.com/devstream-io/devstream/pkg/util/file"
```
Comment (daniel-hutao) [inline internal/pkg/pluginengine/cmd_apply.go:7]: goimports

File: internal/pkg/pluginengine/cmd_delete.go [hunk 4]
```diff
@@ -5,6 +5,8 @@ import (
 	"fmt"
 	"os"
 
+	"github.com/devstream-io/devstream/pkg/util/file"
+
```
Comment (daniel-hutao) [inline internal/pkg/pluginengine/cmd_delete.go:9]: goimports

File: internal/pkg/pluginengine/cmd_destroy.go [hunk 5]
```diff
@@ -5,6 +5,8 @@ import (
 	"fmt"
 	"os"
 
+	"github.com/devstream-io/devstream/pkg/util/file"
+
```
Comment (daniel-hutao) [inline internal/pkg/pluginengine/cmd_destroy.go:9]: goimports

File: internal/pkg/show/status/status.go [hunk 6]
```diff
@@ -5,6 +5,8 @@ import (
 	"reflect"
 	"strings"
 
+	"github.com/devstream-io/devstream/pkg/util/file"
+
```
Comment (daniel-hutao) [inline internal/pkg/show/status/status.go:9]: goimports

**Review by daniel-hutao — CHANGES_REQUESTED** [review]
Good job!

Please test this pr fully, as it is a very basic feature, to avoid introducing more serious bugs.

**Review by daniel-hutao — APPROVED** [review]

**Review by IronCore864 — APPROVED** [review]

**Pull request merged**